### PR TITLE
Improve Xamarin Forms docs

### PR DESCRIPTION
**Pull request by FinHorsley**
**What are the main goals of this change?**
- [x] Better readability (style, grammar, spelling...)
- [ ] Content correction (accuracy, wrong information...)
- [ ] New content

### CODE REVIEW

File: input/docs/handbook/data-binding/xamarin-forms.md [hunk 1]
```diff
@@ -67,7 +79,43 @@ public partial class TheContentPage : ReactiveContentPage<TheViewModel>
...
+                //Don't pass the EventArgs to the command
+                .Select(_ => Unit.Default)
+                .InvokeCommand(ViewModel, x => x.TheTextCommand)
+                .DisposeWith(disposable)p
+        });
+    }
+}
+```
+
+### Performance Tips
```
Comment (glennawatson) [inline input/docs/handbook/data-binding/xamarin-forms.md:114]: On xamarin forms this often is counter to our recommendations. Eg disposing can have a performance cost and may not always be needed. 

Eg this is more about preventing memory leaks and there are scenarios when you don't need to dispose.

File: input/docs/handbook/data-binding/xamarin-forms.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
-For Xamarin.Forms applications, you need to install the `ReactiveUI.XamForms` package and use base classes for your Views from there. Your `ContentPage` should inherit from `ReactiveContentPage<TViewModel>`, your `TextCell` should inherit from `ReactiveTextCell<TViewModel>`, etc. Those classes contain the `IViewFor<TViewModel>` interface implementation. Also, always dispose bindings via [WhenActivated](../when-activated), or else the bindings leak memory.
+For Xamarin.Forms applications you need to install the ReactiveUI.XamForms [Nuget package](https://www.nuget.org/packages/ReactiveUI.XamForms/).
```
Comment (RLittlesII) [inline input/docs/handbook/data-binding/xamarin-forms.md:1]: Maybe we should link to the README where we have a matrix of packages instead?  Only though being, a lot of issues we address have to do with people not knowing which packages to install to which projects.

Comment (RLittlesII) [reply input/docs/handbook/data-binding/xamarin-forms.md:1]: https://github.com/reactiveui/reactiveui#nuget-packages

File: input/docs/handbook/data-binding/xamarin-forms.md [hunk 3]
```diff
@@ -67,7 +79,43 @@ public partial class TheContentPage : ReactiveContentPage<TheViewModel>
 }
 ```
 
-# Routing
+The ReactiveUI binding engine is very powerful, but there may be cases where you want to bind to C# Events coming from your view (such as when a buttons `Tap` event fires). For this scenario use [Pharmacist](https://github.com/reactiveui/Pharmacist), which automatically creates Observables for all the C# Events in your project.
```
Comment (RLittlesII) [inline input/docs/handbook/data-binding/xamarin-forms.md:82]: It's also worth noting 
- ReactiveUI bindings don't use the Xamarin.Forms XAML Binding Engine
- XAML is still an option, you don't need to use ReactiveUI bindings
- You can use both in the same application

File: input/docs/handbook/data-binding/xamarin-forms.md (hunk 2 above)
Comment (FinHorsley) [reply input/docs/handbook/data-binding/xamarin-forms.md:1]: For Xamarin.Forms projects you only need to install the ReactiveUI.XamForms package into the Shared library right? There's no need to install it in each of the platform projects?

File: input/docs/handbook/data-binding/xamarin-forms.md (hunk 1 above)
Comment (FinHorsley) [reply input/docs/handbook/data-binding/xamarin-forms.md:114]: Ah ok. So `WhenActivated` is more about providing an easy way to control the lifetime of a binding (only needs to be disposed when needed to improve performance), as well as preventing memory leaks?

File: input/docs/handbook/data-binding/xamarin-forms.md (hunk 2 above)
Comment (RLittlesII) [reply input/docs/handbook/data-binding/xamarin-forms.md:1]: Correct.  We get a lot of confusion around what package is supposed to be installed in which project.  So keeping them local to our docs for instruction helps in my mind.

Actually this is most likely where we want to link. 
https://reactiveui.net/docs/getting-started/installation/

File: input/docs/handbook/data-binding/xamarin-forms.md (hunk 1 above)
Comment (glennawatson) [reply input/docs/handbook/data-binding/xamarin-forms.md:116]: Kind of, `WhenActivated` and disposing observables isn't more performant, just more correct in certain situations. Eg on WPF/UWP you always have to Dispose due to a potential memory leak which was the motivation of introducing it.

Observable derived from events have the same lifetime rules as events. 

Eg if the current object is the owner of the memory then the GC can free the object. If the current object doesn't own it then the GC can't free the object.

Comment (FinHorsley) [reply input/docs/handbook/data-binding/xamarin-forms.md:116]: So for Xamarin.Forms what is the correct way to use `WhenActivated` 🙂

File: input/docs/handbook/data-binding/xamarin-forms.md (hunk 2 above)
Comment (FinHorsley) [reply input/docs/handbook/data-binding/xamarin-forms.md:1]: Are the ReactiveUI packages self-contained, or are there cases where the core libraries use bait-and-switch to tap into the platform specific code?

Comment (glennawatson) [reply input/docs/handbook/data-binding/xamarin-forms.md:1]: Mixture of both and dynamic DLL leading of known packages.

File: input/docs/handbook/data-binding/xamarin-forms.md (hunk 1 above)
Comment (glennawatson) [reply input/docs/handbook/data-binding/xamarin-forms.md:116]: https://reactiveui.net/docs/handbook/when-activated/#when-should-i-bother-disposing-of-idisposable-objects

Comment (FinHorsley) [reply input/docs/handbook/data-binding/xamarin-forms.md]: I've removed the performance tips that i wrote, i'll re-add my words when i've looked into the performance a bit more

File: input/docs/handbook/data-binding/xamarin-forms.md (hunk 3 above)
Comment (FinHorsley) [reply input/docs/handbook/data-binding/xamarin-forms.md:84]: Added that in

**Pull request merged**